apps/next15-demo/src/app/components/BridgeHandler.tsx

```diff
@@ -0,0 +1,33 @@
+"use client";
+
+import { resolveRelations } from "@/lib/getStory";
+import { StoryblokBridgeV2 } from "@storyblok/js";
+import { useEffect } from "react";
+import { useSWRConfig } from "swr";
+
+declare global {
+  interface window {
+    StoryblokBridge: StoryblokBridgeV2;
+  }
+}
+
+export const BridgeHandler = () => {
+  const { mutate } = useSWRConfig();
+
+  useEffect(() => {
+    const bridge = new window.StoryblokBridge({
+      resolveRelations: resolveRelations,
+    });
+
+    bridge.on("input", (event) => {
+      if (!event || !event.story) return;
+      mutate(`cdn/stories/${event.story.slug}`, event.story, {
+        revalidate: false,
+      });
+    });
```

Comment:
```suggestion
  const cb = event => mutate(`cdn/stories/${event.story.slug}`, event.story, {
    revalidate: false,
  });
  
  useStoryblokBridge(story.id, cb, { resolveRelations });
```

(would need BridgeHandler to have the `story` from `[[...slug]]/page.tsx` to be passed as property)